src/include/stir/ProjDataInfoGeneric.inl

```diff
@@ -83,8 +83,22 @@ ProjDataInfoGeneric::get_tantheta(const Bin& bin) const
   CartesianCoordinate3D<float> _p1;
   CartesianCoordinate3D<float> _p2;
   find_cartesian_coordinates_of_detection(_p1, _p2, bin);
+
+  _p1.z() += z_shift.z();
+  _p2.z() += z_shift.z();
+
+  // do the below transformation to check whether the points are swapped when phi is obtained
+  // if they are swapped theta needs to be inverted, otherwise the oblique segments contain flipped streaks
+  LORAs2Points<float> lor_as_2_points(_p1, _p2);
+  const double R = sqrt(fmax(square(_p1.x()) + square(_p1.y()), square(_p2.x()) + square(_p2.y())));
+  LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
+  lor_as_2_points.change_representation(lor, R);
```

Comment:
now that we have `lor`, can't we just use `lor.get_tantheta()` (or whatever it's called). I think it was a mistake to put this geometry here, as it's already in the LOR code already

Reply:
I haven't found any mention of theta outside ProjDataInfo files. Can you point me to the LOR function you mean?

Reply:
right, that's because it doesn't exist. sorry.

what about then doing
```
	LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
	get_LOR(lor, bin);
```
directly without going via ` find_cartesian_coordinates_of_detection` (presumably, internally `get_LOR` does that anyway). Possibly we then have
```
       return (lor.z1() - lor.z2()) / lor.radius();
```
(I think we might have to define `radius()` in `LORInAxialAndNoArcCorrSinogramCoordinates`)

Reply:
```
LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
get_LOR(lor, bin);

return (lor.z2() - lor.z1()) / (2 * lor.radius());
```
does the trick, i.e. the inverse and using double of the radius. Is this definitely how the radius is defined on the LOR? Couldn't find the place where this conversion happens.

Reply:
Perfect!